R/tm_g_scatterplotmatrix.R

```diff
@@ -15,27 +15,107 @@
 #'
 #' @note For more examples, please see the vignette "Using scatterplot matrix" via
 #'   \code{vignette("using-scatterplot-matrix", package = "teal.modules.general")}.
-#' @export
 #'
 #' @examples
-#' # Scatterplot matrix of variables from ADSL dataset
+#' # general data example
+#' data <- teal_data()
+#' data <- within(data, {
+#'   countries <- data.frame(
+#'     id = c("DE", "FR", "IT", "ES", "PT", "GR", "NL", "BE", "LU", "AT"),
+#'     government = factor(
+#'       c(2, 2, 2, 1, 2, 2, 1, 1, 1, 2),
+#'       labels = c("Monarchy", "Republic")
+#'     ),
+#'     language_family = factor(
+#'       c(1, 3, 3, 3, 3, 2, 1, 1, 3, 1),
+#'       labels = c("Germanic", "Hellenic", "Romance")
+#'     ),
+#'     population = c(83, 67, 60, 47, 10, 11, 17, 11, 0.6, 9),
+#'     area = c(357, 551, 301, 505, 92, 132, 41, 30, 2.6, 83),
+#'     gdp = c(3.4, 2.7, 2.1, 1.4, 0.3, 0.2, 0.7, 0.5, 0.1, 0.4),
+#'     debt = c(2.1, 2.3, 2.4, 2.6, 2.3, 2.4, 2.3, 2.4, 2.3, 2.4)
+#'   )
+#'   sales <- data.frame(
+#'     id = 1:50,
+#'     country_id = sample(
+#'       c("DE", "FR", "IT", "ES", "PT", "GR", "NL", "BE", "LU", "AT"),
+#'       size = 50,
+#'       replace = TRUE
+#'     ),
+#'     year = sort(sample(2010:2020, 50, replace = TRUE)),
+#'     venue = sample(c("small", "medium", "large", "online"), 50, replace = TRUE),
+#'     cancelled = sample(c(TRUE, FALSE), 50, replace = TRUE),
+#'     quantity = rnorm(50, 100, 20),
+#'     costs = rnorm(50, 80, 20),
+#'     profit = rnorm(50, 20, 10)
+#'   )
+#' })
+#' datanames(data) <- c("countries", "sales")
+#' join_keys(data) <- join_keys(
+#'   join_key("countries", "countries", "id"),
+#'   join_key("sales", "sales", "id"),
+#'   join_key("countries", "sales", c("id" = "country_id"))
+#' )
 #'
+#' app <- init(
+#'   data = data,
+#'   modules = modules(
+#'     tm_g_scatterplotmatrix(
+#'       label = "Scatterplot matrix",
+#'       variables = list(
+#'         data_extract_spec(
+#'           dataname = "countries",
+#'           select = select_spec(
+#'             label = "Select variables:",
+#'             choices = variable_choices(data[["countries"]]),
+#'             selected = c("area", "gdp", "debt"),
+#'             multiple = TRUE,
+#'             ordered = TRUE,
+#'             fixed = FALSE
+#'           )
+#'         ),
+#'         data_extract_spec(
+#'           dataname = "sales",
+#'           filter = filter_spec(
+#'             label = "Select variable:",
+#'             vars = "country_id",
+#'             choices = value_choices(data[["sales"]], "country_id"),
+#'             selected = c("DE", "FR"),
```

Comment:
It is possible that countries will filter row so that it will be n<10. I think you can select all on init or add more countries explicitly

Reply:
Done :+1: